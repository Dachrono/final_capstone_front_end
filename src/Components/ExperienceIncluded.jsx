import PropTypes from 'prop-types';

function ExperienceIncluded({ detailsIncluded, mediaStyles }) {
  return (
    <div className={`experience-extra-details w-100 ${mediaStyles}`}>
      <h3 className="experience-extra-details-title">
        Â¿What does this experience include?
      </h3>
      <p className="experience-included w-100">
        { !detailsIncluded
          ? (<span className="no-details"><i>No details included</i></span>)
          : (detailsIncluded)}
      </p>
    </div>
  );
}

ExperienceIncluded.propTypes = {
  detailsIncluded: PropTypes.string.isRequired,
  mediaStyles: PropTypes.string.isRequired,
};

export default ExperienceIncluded;
