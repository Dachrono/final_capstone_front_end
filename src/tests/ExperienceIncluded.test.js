import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import ExperienceIncluded from '../Components/ExperienceIncluded';

describe('The ExperienceIncluded component', () => {
  test('should render correctly into the DOM', async () => {
    const details = 'This is a testing details data';
    const media = 'desktop';

    const { experienceIncludedComponent } = render(
      <ExperienceIncluded mediaStyles={media} detailsIncluded={details} />,
    );
    expect(experienceIncludedComponent).toMatchSnapshot();
  });

  test('should render correctly the experience details', async () => {
    const details = 'This is a testing details data';
    const media = 'desktop';

    render(
      <ExperienceIncluded mediaStyles={media} detailsIncluded={details} />,
    );

    const expectedResult = screen.getByText(/This is a testing details data/i);

    expect(expectedResult).toBeInTheDocument();
  });
});
