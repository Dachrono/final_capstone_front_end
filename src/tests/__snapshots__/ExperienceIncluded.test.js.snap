// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ExperienceIncluded component should render correctly into the DOM 1`] = `undefined`;
